### PR TITLE
xz, gh: deny network access

### PR DESCRIPTION
**Pull request by alebcay**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Expected to fail without https://github.com/Homebrew/brew/pull/17081.

### CODE REVIEW

**Review by SMillerDev — APPROVED** [review]

File: Formula/x/xz.rb [hunk 1]
```diff
@@ -23,6 +23,8 @@ class Xz < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "0736983b952c5273bb5a345008bac7311c2f4b60758d69cc05495d5b050f88f1"
   end
 
+  deny_network_access
```
Comment (MikeMcQuaid) [inline Formula/x/xz.rb:26]:
```suggestion
  deny_network_access!
```

maybe feels nicer?

**Review by MikeMcQuaid — COMMENTED** [review]
Thanks @alebcay!

Would be nice to see a usage of when this also e.g. only denies some methods.

**Comment by SMillerDev** [comment]
I'd suggest blocking cURL from network in the build step and not in the others

**Comment by MikeMcQuaid** [comment]
> Expected to fail without [Homebrew/brew#17081](https://github.com/Homebrew/brew/pull/17081).

It was merged so rerunning CI now.

**Comment by SMillerDev** [comment]
I guess the stable tap_syntax is cancelling the normal one

**Comment by alebcay** [comment]
Indeed, looks like test-bot is using latest brew tag (not latest master) so this CI will not pass until the aforementioned PR lands in a tag.

**Comment by SMillerDev** [comment]
I think that label only applies to OS testing, we need the stable syntax check to be `continue-on-error` to fix this

**Comment by MikeMcQuaid** [comment]
> I think that label only applies to OS testing, we need the stable syntax check to be `continue-on-error` to fix this

Should be able to have it just not kill the main tap syntax job without a fail fast in the matrix or similar.

**Comment by SMillerDev** [comment]
> Should be able to have it just not kill the main tap syntax job without a fail fast in the matrix or similar.

Since I don't see any reason to ever kill the main syntax job because of the stable one, I ignored this and made a fix with my comment: https://github.com/Homebrew/homebrew-core/pull/169867

**Comment by alebcay** [comment]
N.B.: resources used by the `test` block that need download will fail when network access is blocked; unlike for package build, where resources are downloaded before entering the sandbox, the download for test resources occur _inside_ the sandbox.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Looks good, let's try this out!

> N.B.: resources used by the `test` block that need download will fail when network access is blocked; unlike for package build, where resources are downloaded before entering the sandbox, the download for test resources occur _inside_ the sandbox.

I think there's probably a DSL addition required to indicate a test-only resource and `brew fetch --only-test` or similar can download it in advance.

**Pull request merged**